var/spack/repos/builtin/packages/tensorflow/package.py

```diff
@@ -0,0 +1,291 @@
+    depends_on('py-protobuf@3.3.0:',  type=('build', 'run'), when='@1.3.0:1.6.0')        # noqa: E501
+    depends_on('py-protobuf@3.6.0',   type=('build', 'run'), when='@1.8.0:')
+    depends_on('py-setuptools',       type=('build', 'run'))
+    depends_on('py-six@1.10.0:',      type=('build', 'run'))
+    depends_on('py-termcolor@1.1.0:', type=('build', 'run'), when='@1.6.0:')
+    depends_on('py-wheel',            type=('build', 'run'))
+
+    depends_on('cuda', when='+cuda')
+    depends_on('cudnn', when='+cuda')
+    depends_on('nccl', when='+nccl')
+
+    patch('url-zlib.patch',  when='@0.10.0')
+    patch('crosstool.patch', when='@1.0.0-rc2')  # also on 0.10.0 if +cuda!
+    patch('io_bazel_rules_docker2.patch', when='@1.15.0,2.0.0')
+    patch('http_archive.patch', when='@1.12.3')
+
+
+    def install(self, spec, prefix):
+        if '+gcp' in spec:
+            env['TF_NEED_GCP'] = '1'
+        else:
+            env['TF_NEED_GCP'] = '0'
+
+        env['PYTHON_BIN_PATH'] = str(spec['python'].prefix.bin) + '/python'
+        env['SWIG_PATH'] = str(spec['swig'].prefix.bin)
+        env['GCC_HOST_COMPILER_PATH'] = spack_cc
+
+        # CUDA related config options - note: tf has only been tested for cpu
+        if '+cuda' in spec:
+            env['GCC_HOST_COMPILER_PATH'] = self.compiler.cc #TODO double check if this is still necessary
+            env['TF_NEED_CUDA'] = '1'
+            env['TF_CUDA_VERSION'] = str(spec['cuda'].version.up_to(2))
+            env['TF_CUDNN_VERSION'] = str(spec['cudnn'].version)[0]
+            # TODO also consider the case of cuda enabled but nccl disabled
+            env['TF_NCCL_VERSION'] = str(spec['nccl'].version.up_to(1))
+            if self.spec.satisfies('@1.14.0:'):
+                env['TF_CUDA_PATHS'] = '"' + str(spec['cuda'].prefix)+',' +\
+                                        str(spec['nccl'].prefix)+',' +\
+                                        str(spec['cudnn'].prefix)+'"'
+            env['CUDA_TOOLKIT_PATH'] = str(spec['cuda'].prefix)
+            env['CUDNN_INSTALL_PATH'] = str(spec['cudnn'].prefix) # ignored? as of tf@1.14.0:
+            # TODO: create a string valued variant for compute capabilities?
+            # one should be able to specify single or multiple capabilities
+            env['TF_CUDA_COMPUTE_CAPABILITIES'] = "6.1,7.5"
+            env['NCCL_INSTALL_PATH'] = str(spec['nccl'].prefix.lib)
+            env['NCCL_HDR_PATH'] = str(spec['nccl'].prefix.include)
+            env['TF_CUDA_CLANG'] = '0'
+            env['TF_NEED_ROCM'] = '0'
+            env['TF_NEED_TENSORRT'] = '0'
+        else:
+            env['TF_NEED_CUDA'] = '0'
+            env['TF_CUDA_VERSION'] = ''
+            env['CUDA_TOOLKIT_PATH'] = ''
+            env['TF_CUDNN_VERSION'] = ''
+            env['CUDNN_INSTALL_PATH'] = ''
+
+        # config options for version 1.0
+        if self.spec.satisfies('@1.0.0-rc2:'):
+            env['CC_OPT_FLAGS'] = '-march=x86-64 -mtune=generic'
```

Comment:
This depends on the compiler, and Spack already injects these things with its microarch detection. Remove this.